sermant-plugins/sermant-tag-transmission/tag-transmission-rest-feign-plugin/src/main/java/io/sermant/tag/transmission/rpc/interceptors/ClientHttpRequestTagTransmissionInterceptor.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (C) 2024-2024 Sermant Authors. All rights reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package io.sermant.tag.transmission.rpc.interceptors;
+
+import io.sermant.core.common.LoggerFactory;
+import io.sermant.core.plugin.agent.entity.ExecuteContext;
+import io.sermant.core.plugin.agent.interceptor.AbstractInterceptor;
+import io.sermant.core.utils.CollectionUtils;
+import io.sermant.core.utils.tag.TrafficUtils;
+import io.sermant.tag.transmission.config.strategy.TagKeyMatcher;
+
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpRequest;
+
+import java.util.List;
+import java.util.Map;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * ClientHttpRequestInterceptor enhancement class, initiate restTemplate request method
+ *
+ * @author chengyouling
+ * @since 2024-12-30
+ */
+public class ClientHttpRequestTagTransmissionInterceptor extends AbstractInterceptor {
+    private static final Logger LOGGER = LoggerFactory.getLogger();
```

Comment:
implement AbstractClientInterceptor and make code more clear

Reply:
delete